include/xtensor/xaccessible.hpp

```diff
@@ -212,6 +220,24 @@ namespace xt
         return derived_cast()(static_cast<size_type>(args)...);
     }
 
+    /**
+     * Returns a constant reference to first the element of the expression
+     */
+    template <class D>
+    inline auto xconst_accessible<D>::front() const -> const_reference
+    {
+        return derived_cast().front();
```

Comment:
This means you must implement the `front` and `back` methods in all inheriting classes. And in this case, you don't need to put it in the base class.

Reply:
Sorry I didn't get it ;)